File: 09-02-2024_check_for_children_sum_property_in_a_binary_tree.cpp
class Solution{
    public:
    int isSumProperty(Node *root)
    {
     if(root==NULL) return 1;
     
     if(root->left==NULL && root->right==NULL) return 1;
     
     int sum=0;
     
     if(root->left!=NULL) sum+=root->left->data;
     if(root->right!=NULL) sum+=root->right->data;
     
     return (sum==root->data && isSumProperty(root->left) && isSumProperty(root->right)) ;
    
    }
};

File: 09-02-2024_check_for_children_sum_property_in_a_binary_tree.java
class Solution
{
   
    public static int isSumProperty(Node root) {
      
        if (root == null)
            return 1;

        if (root.left == null && root.right == null)
            return 1;

        int sum = 0;

        if (root.left != null)
            sum += root.left.data;
        if (root.right != null)
            sum += root.right.data;

        return (sum == root.data && isSumProperty(root.left) == 1 && isSumProperty(root.right) == 1) ? 1 : 0;
    }
}
